[WASI-NN] ggml: support load_by_name_with_config

dm4 opened the pull request:
- When the graph builder length > 1, the data of builder[1] is the metadata.
- The metadata is a json string as before.
- Still support set metadata from set_input with index 1 for backward compatibility.
- Also support pass model parameter from cli (only support `n-gpu-layers` currently). For example: `--nn-preload default:GGML:AUTO:llama-2-7b-chat.Q5_K_M.gguf,ngl=99`.
- Move parameters to `struct Graph` and move `LlamaContext` to `struct Context`.

juntao commented:
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

dm4 commented:
Because I have made modifications to the `mdBuild` related content, I would also like to ask @CaptainVincent for help in reviewing it. Thank you!

CaptainVincent reviewed (dismissed):
LGTM. I'm primarily reviewing CL b1a255c.

dm4 commented:
@hydai according to our previous review, I have updated:
- Moved model config parsing into a helper function.
- Used `emplace_back`
- Used `const uint32_t`
